### PR TITLE
chore(CPlusPlus): add detecting a cycle in a graph

### PR DESCRIPTION
**Pull request by kartik-ey1**
This is a slightly Different algorithm to find cycle in directed graph

### CODE REVIEW

**Comment by kartik-ey1** [comment]
I didnt understand the error in this everything is working fine in the code

File: algorithms/CPlusPlus/Graphs/detecting-cycle-in-a-graph-using-three-color-mechanism.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	//  DFS traversal of the vertices reachable from v
+    	bool DFSUtil(int v, int color[]);
+	
+public:
+
+	Graph(int V); // The Constructor
+
+  // Function to add an edge to graph
+	void addEdge(int v, int w);
+ // Function to check wheather graph is cyclic or not
```
Comment (mohitchakraverty) [inline algorithms/CPlusPlus/Graphs/detecting-cycle-in-a-graph-using-three-color-mechanism.cpp:29]:
```suggestion
 // Function to check whether graph is cyclic or not
```

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Please fix the typo

File: algorithms/CPlusPlus/Graphs/detecting-cycle-in-a-graph-using-three-color-mechanism.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		if (DFSUtil(i, color) == true)
+			return true;
+
+	return false;
+}
+
+// Driver code to test above
+int main()
+{
+	// Create a graph given in the above diagram
```
Comment (mohitchakraverty) [inline algorithms/CPlusPlus/Graphs/detecting-cycle-in-a-graph-using-three-color-mechanism.cpp:102]: Please also show the graph diagrammatically which you have considered for better understanding of users.

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]

File: algorithms/CPlusPlus/Graphs/detecting-cycle-in-a-graph-using-three-color-mechanism.cpp (hunk 2 above)
Comment (kartik-ey1) [reply algorithms/CPlusPlus/Graphs/detecting-cycle-in-a-graph-using-three-color-mechanism.cpp:112]: Done

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Update your respective README.md file

**Comment by kartik-ey1** [comment]
Okay , will make a new folder and will add a readme fole with this same codefile into it 
Thank you

**Comment by mohitchakraverty** [comment]
No, just update the README file with name and link under correct topic
In your case it will be - algorithms/CPlusPlus/README.md

**Review by mohitchakraverty — APPROVED** [review]
Looks great !

**Pull request merged**